### ISSUE TITLE
Deprecated calls

### ISSUE THREAD

**Issue by ProblematicDude** [miguelpruivo/flutter_file_picker#653]
**Describe the bug**
While building the apk with ```flutter build apk``` it gives me warning that a deprecated API is used. 
Can you please update the plugin ?

```file picker version file_picker: ^3.0.0```
**Platform**
-  Android

[This is now deprecated](https://api.flutter.dev/javadoc/io/flutter/plugin/common/PluginRegistry.Registrar.html)

**Error Log**
```
2 warnings
E:\Software\IDEs\Android\flutter\.pub-cache\hosted\pub.dartlang.org\file_picker-3.0.0\android\src\main\java\com\mr\flutter\plugin\filepicker\FilePickerPlugin.java:26: warning: [depreca
tion] Registrar in PluginRegistry has been deprecated
import io.flutter.plugin.common.PluginRegistry.Registrar;
                                              ^
E:\Software\IDEs\Android\flutter\.pub-cache\hosted\pub.dartlang.org\file_picker-3.0.0\android\src\main\java\com\mr\flutter\plugin\filepicker\FilePickerPlugin.java:120: warning: [deprec
ation] Registrar in PluginRegistry has been deprecated
    public static void registerWith(final Registrar registrar) {

```

**Screenshots and/or video**
![studio64_aqScBquoS6](https://user-images.githubusercontent.com/59223300/111257233-c705b100-8640-11eb-94f2-2b3c0441244a.png)

**Comment by andrewknockin** [comment]
2 warnings are now 3. Using current Gradle v7.0 or higher
Line 248 also flags the deprecated call
.../flutter_file_picker/android/src/main/java/com/mr/flutter/plugin/filepicker/FilePickerPlugin.java:248: warning: [deprecation] Registrar in PluginRegistry has been deprecated
            final PluginRegistry.Registrar registrar,
                                ^
This actually prevents my project using file_picker compiling with the latest Android SDK

**Comment by miguelpruivo** [comment]
@andrewknockin how does that prevent you from building?

**Comment by andrewknockin** [comment]
Miguel, Thanks for following up promptly, and thanks for your code.
When compiling our code, I get one error as shown below. Its a Flutter project with many files which my colleague can successfully compile, but downloading his code from GitHub gives me this error. We have looked in detail but cannot spot the problem in our different environments. However, the error message says "failed likely due to AndroidX incompatibilities in a plugin" for file_picker3.0.0, and "Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0."
My environment is:
 Flutter (Channel stable, 2.0.6, on Mac OS X 10.15.7 19H1030 darwin-x64, locale en-GB)
Android toolchain - develop for Android devices (Android SDK version 30.0.3)
Android Studio (version 4.2)

[+1001 ms] > Task :app:processDebugResources FAILED
[        ] Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
[        ] Use '--warning-mode all' to show the individual deprecation warnings.
[        ] See https://docs.gradle.org/6.1.1/userguide/command_line_interface.html#sec:command_line_warnings
[        ] 60 actionable tasks: 5 executed, 55 up-to-date
[   +1 ms] FAILURE: Build failed with an exception.
[   +3 ms] * What went wrong:
[        ] Execution failed for task ':app:processDebugResources'.
[        ] > A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
[        ]    > Android resource linking failed
[        ]      /Users/andrew/dev/TGM/la/build/file_picker/intermediates/library_manifest/debug/AndroidManifest.xml:9:5-15:15: AAPT: error: unexpected element <queries> found in <manifest>.
[        ] * Try:
[        ] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[        ] * Get more help at https://help.gradle.org
[        ] BUILD FAILED in 22s
[{"event":"app.progress","params":{"appId":"7c04d720-4895-40cb-b4c8-644dd565180e","id":"0","progressId":null,"finished":true}}]
[ +655 ms] The build failed likely due to AndroidX incompatibilities in a plugin. The tool is about to try using Jetifier to solve the incompatibility.
[   +8 ms] Building plugin file_picker...
Running Gradle task 'assembleAarRelease'...
[{"event":"app.progress","params":{"appId":"7c04d720-4895-40cb-b4c8-644dd565180e","id":"1","progressId":null,"message":"Running Gradle task 'assembleAarRelease'...","finished":false}}]
[   +8 ms] Using gradle from /Users/andrew/dev/flutter/.pub-cache/hosted/pub.dartlang.org/file_picker-3.0.0/android/gradlew.
[        ] /Users/andrew/dev/flutter/.pub-cache/hosted/pub.dartlang.org/file_picker-3.0.0/android/gradlew mode: 33261 rwxr-xr-x.
[        ] executing: [/Users/andrew/dev/flutter/.pub-cache/hosted/pub.dartlang.org/file_picker-3.0.0/android/] /Users/andrew/dev/flutter/.pub-cache/hosted/pub.dartlang.org/file_picker-3.0.0/android/gradlew -I=/Users/andrew/dev/flutter/packages/flutter_tools/gradle/aar_init_script.gradle -Pflutter-root=/Users/andrew/dev/flutter -Poutput-dir=/Users/andrew/dev/TGM/la/build/app -Pis-plugin=true -PbuildNumber=1.0 -Pverbose=true -Ptarget-platform=android-arm,android-arm64,android-x64 assembleAarRelease
[        ] 
           FAILURE: Build failed with an exception.

           * What went wrong:
           A problem occurred configuring root project 'file_picker'.
           > SDK location not found. Define location with sdk.dir in the local.properties file or with an ANDROID_HOME environment variable.

           * Try:
           Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

           * Get more help at https://help.gradle.org

           BUILD FAILED in 7s

[   +7 ms] Exception: The plugin file_picker could not be built due to the issue above.

**Comment by andrewknockin** [comment]
Now it compiles and runs.
I found this webpage
[https://android-developers.googleblog.com/2020/07/preparing-your-build-for-package-visibility-in-android-11.html](https://android-developers.googleblog.com/2020/07/preparing-your-build-for-package-visibility-in-android-11.html)
which says to edit the android > build.gradle file and change line 9 which declares the classpath for the gradle build version to ensure the dot release is the upgrade version. In my case I changed 3.5.0 to 3.5.4 and this removes the linking failure because of the unexpected element being found.
Thanks again for your code.
